From 64a180ce4296dd6ac727ad14eae55d8073a2e2b2 Mon Sep 17 00:00:00 2001
From: Mike FABIAN <mfabian@redhat.com>
Date: Mon, 6 Nov 2017 16:23:27 +0100
Subject: [PATCH] Make "tw" the default keyboard layout for zh_TW and cmn_TW

Resolves: rhbz#1387825 - The default keyboard layout for zh_TW (traditional Chinese) should be "us" instead of "cn"
See: https://bugzilla.redhat.com/show_bug.cgi?id=1387825
---
 data/languages.xml   |  4 ++--
 data/territories.xml |  2 +-
 test_cases.py        | 16 ++++++++++++----
 3 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/data/languages.xml b/data/languages.xml
index a2656c8..d0d2dd4 100644
--- a/data/languages.xml
+++ b/data/languages.xml
@@ -4153,7 +4153,7 @@
       <territory><territoryId>TW</territoryId><rank>1000</rank></territory>
     </territories>
     <keyboards>
-      <keyboard><keyboardId>cn</keyboardId><rank>1000</rank></keyboard>
+      <keyboard><keyboardId>tw</keyboardId><rank>1000</rank></keyboard>
     </keyboards>
     <inputmethods>
       <inputmethod><inputmethodId>ibus/libpinyin</inputmethodId><rank>900</rank></inputmethod>
@@ -30209,7 +30209,7 @@
       <territory><territoryId>TW</territoryId><rank>1000</rank></territory>
     </territories>
     <keyboards>
-      <keyboard><keyboardId>cn</keyboardId><rank>1000</rank></keyboard>
+      <keyboard><keyboardId>tw</keyboardId><rank>1000</rank></keyboard>
     </keyboards>
     <inputmethods>
       <inputmethod><inputmethodId>ibus/chewing</inputmethodId><rank>1000</rank></inputmethod>
diff --git a/data/territories.xml b/data/territories.xml
index 2943414..5e25be3 100644
--- a/data/territories.xml
+++ b/data/territories.xml
@@ -22788,7 +22788,7 @@
       <language><languageId>zh</languageId><rank>1000</rank></language>
     </languages>
     <keyboards>
-      <keyboard><keyboardId>cn</keyboardId><rank>1000</rank></keyboard>
+      <keyboard><keyboardId>tw</keyboardId><rank>1000</rank></keyboard>
     </keyboards>
     <inputmethods>
       <inputmethod><inputmethodId>ibus/chewing</inputmethodId><rank>1000</rank></inputmethod>
diff --git a/test_cases.py b/test_cases.py
index a53ba10..68e7ea6 100644
--- a/test_cases.py
+++ b/test_cases.py
@@ -726,8 +726,16 @@ def dummy():
         TW: ['zh_TW.UTF-8']
          +: ['zh_TW.UTF-8']
         zh: ['cn']
-        TW: ['cn']
-         +: ['cn']
+        TW: ['tw']
+         +: ['tw']
+
+    >>> _test_language_territory(show_weights=False, languageId="cmn", territoryId="TW") # doctest: +NORMALIZE_WHITESPACE
+        cmn: ['cmn_TW.UTF-8']
+        TW: ['zh_TW.UTF-8']
+         +: ['cmn_TW.UTF-8', 'zh_TW.UTF-8']
+        cmn: ['tw']
+        TW: ['tw']
+         +: ['tw']
 
     >>> _test_language_territory(show_weights=False, languageId="zh", territoryId="HK") # doctest: +NORMALIZE_WHITESPACE
         zh: ['zh_CN.UTF-8', 'zh_TW.UTF-8', 'zh_HK.UTF-8', 'zh_SG.UTF-8']
@@ -790,8 +798,8 @@ def dummy():
         TW: ['zh_TW.UTF-8']
          +: ['zh_CN.UTF-8', 'zh_SG.UTF-8', 'zh_TW.UTF-8']
         zh: ['cn']
-        TW: ['cn']
-         +: ['cn']
+        TW: ['tw']
+         +: ['tw', 'cn']
 
     >>> _test_language_territory(show_weights=False, languageId="zh", scriptId = "Hant", territoryId="HK") # doctest: +NORMALIZE_WHITESPACE
         zh: ['zh_CN.UTF-8', 'zh_TW.UTF-8', 'zh_HK.UTF-8', 'zh_SG.UTF-8']
-- 
2.21.0

